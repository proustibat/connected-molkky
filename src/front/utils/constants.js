export default {
  POSITION: {
    KNOCKED_OVER: 'KNOCKED_OVER',
    UPRIGHT: 'UPRIGHT',
  },

  MESSAGE_TYPE: {
    INFO: 'info',
    WARNING: 'warning',
    ERROR: 'error',
  },
};
