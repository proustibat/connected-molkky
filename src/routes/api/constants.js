export const SKITTLES = {
  AC233FA247EB: 1,
  AC233FA247F0: 2,
  AC233FA247F5: 3,
  AC233FA247F6: 4,
  AC233FA246F1: 5,
  AC233FA247FA: 6,
  AC233FA246FE: 7,
  AC233FA24701: 8,
  AC233FA2470C: 9,
  AC233FA2470D: 10,
  AC233FA246FC: 11,
  AC233FA246FD: 12,
};

export const POSITION_LEVEL = 0.8;
