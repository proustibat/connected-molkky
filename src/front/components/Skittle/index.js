import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import SVGCircle from '@components/SVGCircle';
import constants from '@utils/constants';

const { POSITION } = constants;

const Skittle = ({ position, value, size }) => {
  const initialColors = {
    textColor: '#0944ff',
    strokeColor: '#0944ff',
    fillColor: '#d7ffff',
  };
  const lightColor = '#ebebeb';
  const darkColor = '#282828';

  const [textColor, setTextColor] = useState(initialColors.textColor);
  const [strokeColor, setStrokeColor] = useState(initialColors.strokeColor);
  const [fillColor, setFillColor] = useState(initialColors.fillColor);

  useEffect(() => {
    if (position) {
      setTextColor(position === POSITION.UPRIGHT ? darkColor : lightColor);
      setStrokeColor(position === POSITION.UPRIGHT ? darkColor : lightColor);
      setFillColor(position === POSITION.UPRIGHT ? lightColor : darkColor);
    } else {
      setTextColor(initialColors.textColor);
      setStrokeColor(initialColors.strokeColor);
      setFillColor(initialColors.fillColor);
    }
  }, [position]);

  return (
    <div style={{ width: size, height: size }}>
      <SVGCircle
        text={value}
        textColor={textColor}
        strokeColor={strokeColor}
        fillColor={fillColor}
        width="100%"
        height="100%"
      />
    </div>
  );
};

Skittle.propTypes = {
  position: PropTypes.oneOf(Object.values(POSITION)),
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  size: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

Skittle.defaultProps = {
  position: null,
  size: undefined,
};

export default Skittle;
