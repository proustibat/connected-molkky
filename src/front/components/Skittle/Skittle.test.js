import { cleanup, render } from '@testing-library/react';
import React from 'react';
import Skittle from './index';
import constants from '@utils/constants';


describe('Skittle', () => {
  afterEach(cleanup);

  it('should render the component correctly without position prop', () => {
    // Given / When
    const { container } = render(
      <Skittle value="qwerty" />,
    );

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should render the component correctly with upright position prop', () => {
    // Given / When
    const { container } = render(
      <Skittle value="qwerty" position={constants.POSITION.UPRIGHT} />,
    );

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should render the component correctly with knocked over position prop', () => {
    // Given / When
    const { container } = render(
      <Skittle value="qwerty" position={constants.POSITION.KNOCKED_OVER} />,
    );

    // Then
    expect(container).toMatchSnapshot();
  });
});
