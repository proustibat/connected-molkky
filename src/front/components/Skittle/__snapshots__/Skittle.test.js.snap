// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skittle should render the component correctly with knocked over position prop 1`] = `
<div>
  <div>
    <svg
      height="100%"
      viewBox="-25 -25 400 400"
      width="100%"
    >
      <circle
        cx="175"
        cy="175"
        fill="#282828"
        r="175"
        stroke="#ebebeb"
        stroke-width="25"
      />
      <text
        dominant-baseline="central"
        fill="#ebebeb"
        style="font-size: 175px;"
        text-anchor="middle"
        x="175"
        y="175"
      >
        qwerty
      </text>
    </svg>
  </div>
</div>
`;

exports[`Skittle should render the component correctly with upright position prop 1`] = `
<div>
  <div>
    <svg
      height="100%"
      viewBox="-25 -25 400 400"
      width="100%"
    >
      <circle
        cx="175"
        cy="175"
        fill="#ebebeb"
        r="175"
        stroke="#282828"
        stroke-width="25"
      />
      <text
        dominant-baseline="central"
        fill="#282828"
        style="font-size: 175px;"
        text-anchor="middle"
        x="175"
        y="175"
      >
        qwerty
      </text>
    </svg>
  </div>
</div>
`;

exports[`Skittle should render the component correctly without position prop 1`] = `
<div>
  <div>
    <svg
      height="100%"
      viewBox="-25 -25 400 400"
      width="100%"
    >
      <circle
        cx="175"
        cy="175"
        fill="#d7ffff"
        r="175"
        stroke="#0944ff"
        stroke-width="25"
      />
      <text
        dominant-baseline="central"
        fill="#0944ff"
        style="font-size: 175px;"
        text-anchor="middle"
        x="175"
        y="175"
      >
        qwerty
      </text>
    </svg>
  </div>
</div>
`;
